.github/workflows/deploy-opentofu.yml

```diff
@@ -0,0 +1,80 @@
+name: Deploy OpenTofu
+
+on:
+  push:
+    paths:
+      - 'web/deploy/terraform/**'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.2.2
+
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: ${{ inputs.aws_region }}
+          role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCOUNT_ID }}:role/github-actions-role-shared
+
+      - name: Setup OpenTofu
+        uses: opentofu/setup-opentofu@v1
+
+      - name: Check for changes in reusable modules
+        id: changed-tofu-modules
+        uses: tj-actions/changed-files@v45
+        with:
+          files: |
+            web/deploy/terraform/modules/**
+
+      - name: Check for changes in shared deployment
+        id: changed-tofu-shared
+        uses: tj-actions/changed-files@v45
+        with:
+          files: |
+            web/deploy/terraform/shared/**
+
+      - name: Check for changes in staging deployment
+        id: changed-tofu-staging
+        uses: tj-actions/changed-files@v45
+        with:
+          files: |
+            web/deploy/terraform/staging/**
+
+      - name: Check for changes in production deployment
+        id: changed-tofu-production
+        uses: tj-actions/changed-files@v45
+        with:
+          files: |
+            web/deploy/terraform/production/**
+
+      - name: Initialize shared resources
+        run: |
+          tofu -chdir web/deploy/terraform/shared init
+
+      - name: Deploy shared resources
+        if: steps.changed-tofu-modules.outputs.any_modified == true || steps.changed-tofu-shared.outputs.any_modified == true
+        run: |
+          tofu -chdir web/deploy/terraform/shared apply -nocolor -auto-approve
+
+      - name: Initialize staging resources
+        run: |
+          tofu -chdir web/deploy/terraform/staging init
+
+      - name: Deploy staging resources
+        if: steps.changed-tofu-modules.outputs.any_modified == true || steps.changed-tofu-shared.outputs.any_modified == true || steps.changed-tofu-staging.outputs.any_modified == true
+        run: |
+          tofu -chdir web/deploy/terraform/staging apply -nocolor -auto-approve
+
+      - name: Initialize production resources
+        run: |
+          tofu -chdir web/deploy/terraform/staging init
+
+      - name: Deploy production resources
+        if: steps.changed-tofu-modules.outputs.any_modified == true || steps.changed-tofu-shared.outputs.any_modified == true || steps.changed-tofu-production.outputs.any_modified == true
+        run: |
+          tofu -chdir web/deploy/terraform/production apply -nocolor -auto-approve
```

Comment:
We can probably simplify this by using a matrix with the three environments. Something like:
```yaml
strategy:
  matrix:
    environments: ["shared", "staging", "prod"]
```

Reply:
Addressed